pkg/operator/ceph/cluster/utils.go

```diff
@@ -30,28 +30,84 @@ import (
 
-	cm := &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      k8sutil.ConfigOverrideName,
-			Namespace: namespace,
-		},
-		Data: placeholderConfig,
+	existingCM, err := clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Get(ctx, k8sutil.ConfigOverrideName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			logger.Warningf("failed to get cm %q to check labels and annotations", k8sutil.ConfigOverrideName)
+			return nil
+		}
+
+		labels := map[string]string{}
+		annotations := map[string]string{}
+		initRequiredMetadata(clusterMetadata, labels, annotations)
+
+		// Create the configmap since it doesn't exist yet
+		placeholderConfig := map[string]string{
+			k8sutil.ConfigOverrideVal: "",
+		}
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        k8sutil.ConfigOverrideName,
+				Namespace:   namespace,
+				Labels:      labels,
+				Annotations: annotations,
+			},
+			Data: placeholderConfig,
+		}
+
+		err := ownerInfo.SetControllerReference(cm)
+		if err != nil {
+			return errors.Wrapf(err, "failed to set owner reference to override configmap %q", cm.Name)
+		}
+		_, err = clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Create(ctx, cm, metav1.CreateOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to create override configmap %s", namespace)
+		}
+		logger.Infof("created placeholder configmap for ceph overrides %q", cm.Name)
+		return nil
 	}
 
-	err := ownerInfo.SetControllerReference(cm)
-	if err != nil {
-		return errors.Wrapf(err, "failed to set owner reference to override configmap %q", cm.Name)
+	// Ensure the annotations and labels are initialized
+	if existingCM.Annotations == nil {
+		existingCM.Annotations = map[string]string{}
 	}
-	_, err = clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Create(ctx, cm, metav1.CreateOptions{})
-	if err != nil && !kerrors.IsAlreadyExists(err) {
-		return errors.Wrapf(err, "failed to create override configmap %s", namespace)
+	if existingCM.Labels == nil {
+		existingCM.Labels = map[string]string{}
 	}
 
+	// Add recommended labels and annotations to the existing configmap if it doesn't have any yet
+	updateRequired := initRequiredMetadata(clusterMetadata, existingCM.Labels, existingCM.Annotations)
+	if updateRequired {
+		_, err = clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Update(ctx, existingCM, metav1.UpdateOptions{})
+		if err != nil {
+			logger.Warningf("failed to add recommended labels and annotations to configmap %q. %v", existingCM.Name, err)
+		} else {
+			logger.Infof("added expected labels and annotations to configmap %q", existingCM.Name)
+		}
+	}
 	return nil
 }
+
+func initRequiredMetadata(metadata metav1.ObjectMeta, labels, annotations map[string]string) bool {
+	// Add the helm labels in case the user wants to install the cluster helm chart to start managing it
+	// Add the helm annotations in case the user wants to install the cluster helm chart to start managing it
```

Comment:
can we club it together?
```suggestion
	// Add the helm labels and annotations in case the user wants to install the cluster helm chart to start managing it
```